client/my-sites/marketing/tools/feature-buttons-with-plan-gate.tsx

```diff
@@ -0,0 +1,74 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import React, { FunctionComponent } from 'react';
+import page from 'page';
+import { useTranslate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { addQueryArgs } from 'lib/url';
+import Button from 'components/button';
+import { getPlan } from 'lib/plans';
+import { hasFeature } from 'state/sites/plans/selectors';
+import { getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+
+interface MarketingToolsFeatureButtonsWithPlanGateConnectProps {
+	hasPlanFeature: boolean;
+	planTitle: string;
+	selectedSiteSlug: string | null;
+}
+
+interface MarketingToolsFeatureButtonsWithPlanGateOwnProps {
+	buttonText: string;
+	feature: string;
+	handleButtonClick: () => void;
+	minPlanSlug: string;
+}
+
+const MarketingToolsFeatureButtonsWithPlanGate: FunctionComponent<
+	MarketingToolsFeatureButtonsWithPlanGateOwnProps &
+		MarketingToolsFeatureButtonsWithPlanGateConnectProps
+> = ( {
+	buttonText,
+	handleButtonClick,
+	hasPlanFeature,
+	planTitle,
+	minPlanSlug,
+	selectedSiteSlug,
+} ) => {
+	const translate = useTranslate();
+
+	const handleUpgradeClick = () => {
+		page( addQueryArgs( { plan: minPlanSlug }, `/plans/${ selectedSiteSlug }` ) );
```

Comment:
It would be great to be able to redirect the user back to the `Marketing Tools` page once they have purchased a plan, but I'm not sure how easy it is to do.

Reply:
lemme see what we can do